### PR TITLE
Introduced Login class for hashing SHA512 and checking proper hash.

### PR DESCRIPTION
**Pull request by BoskiJanusz**
Login class will be inherided by class Admin to hash new passwords.
    Commit contains basic class and tests. It will be extended by:
     - map that contains questions to reset password
     - tests for private members - proper storing passwords in file (this class should not make files with login and pasword, it should just be able to read them properly)

Modified makefiles and docker for new files.

### CODE REVIEW

File: src/app/login/login.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#include <nlohmann/json.hpp>
+#include <openssl/evp.h>
+#include <sstream>
+#include <string>
+
+class Login
+{
+    std::string user_name_{};
+    std::string password_hash_{};
+    //  std::map<std::string, std::string> mapOfRandomQuestionsAndAnswers{}; To do map with questions to reset password
```
Comment (fbtom) [inline src/app/login/login.hpp:15]: Please mark TODO #number where number will point out to issue on github

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: src/tests/ut/login/login_test.cpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#include "login_test.hpp"
+
+namespace
+{
+TEST_F(LoginTestFixture, GivenHashedSHA512DataComparedToCalculatedHashedData)
+{
+    auto expected_value = "b109f3bbbc244eb82441917ed06d618b9008dd09b3befd1b5e07394c706a8bb980b1d7785e5976ec049b46df5f13"
```
Comment (fbtom) [inline src/tests/ut/login/login_test.cpp:7]: const auto

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @BoskiJanusz for making course materials better!

**Pull request merged**